src/Command/SitemapGeneratorCommand.php

```diff
@@ -56,62 +118,97 @@ public function configure()
      */
     public function execute(InputInterface $input, OutputInterface $output)
     {
-        $this->getContainer()->get('router')->getContext()->setHost($input->getArgument('host'));
-        $this->getContainer()->get('router')->getContext()->setScheme($input->getOption('scheme'));
-        $this->getContainer()->get('router')->getContext()->setBaseUrl($input->getOption('baseurl'));
+        $host = $input->getArgument('host');
+        $scheme = $input->getOption('scheme');
+        $baseUrl = $input->getOption('baseurl');
+        $permanentFolder = $input->getArgument('folder');
+        $appendPath = $input->hasOption('sitemap_path') ? $input->getOption('sitemap_path') : $baseUrl;
+
+        $this->getContext()->setHost($host);
+        $this->getContext()->setScheme($scheme);
+        $this->getContext()->setBaseUrl($baseUrl);
+
+        $tempFolder = $this->createTempFolder($output);
```

Comment:
"folder" is a graphic representation of a filesystem directory.
```suggestion
        $tempDir = $this->createTempFolder($output);
```

Reply:
Good tip! Thanks @phansys !